common/src/main/java/com/microsoft/identity/common/internal/providers/oauth2/WebViewAuthorizationFragment.java

```diff
@@ -459,6 +460,46 @@ public ActivityResultLauncher<LegacyFido2ApiObject> getFidoLauncher() {
         return mFidoLauncher;
     }
 
+    /**
+     * Launches the web browser intent for the given uri.
+     *
+     * @param uri The uri to launch the web browser intent.
+     */
+    public boolean launchWebBrowserIntent(@NonNull final Uri uri) {
+        final String methodTag = TAG + ":launchWebBrowserIntent";
+        if (mAuthIntent != null) {
+            Logger.info(methodTag, "Launching web browser intent for SwitchBrowserProtocol.");
+            mAuthIntent.setData(uri);
+            startActivity(mAuthIntent);
+            return true;
+        }
+        return false;
+    }
+
+    /**
+     * Constructs the switch browser uri with the given action uri and parameters.
+     *
+     * @param action_uri The action uri.
+     * @param params     The parameters.
+     * @return The constructed uri.
+     */
+    public Uri constructSwitchBrowserUri(@NonNull final String action_uri, @Nullable final HashMap<String, String> params) {
+        final String[] paths = action_uri.split("/");
+        final String authority = paths[0];
+        final Uri.Builder uriBuilder = new Uri.Builder()
+                .scheme("https")
+                .encodedAuthority(authority);
+        for (int i = 1; i < paths.length; i++) {
+            uriBuilder.appendPath(paths[i]);
+        }
+        if (params != null) {
+            for (HashMap.Entry<String, String> entry : params.entrySet()) {
+                uriBuilder.appendQueryParameter(entry.getKey(), entry.getValue());
+            }
+        }
+        return uriBuilder.build();
+    }
```

Comment:
do these two methods need to be here? 

I was thinking we can probably create SwitchBrowserRedirectHandler (maybe that extends IChallangeHandler) class and use it receiving a valid switch browser redirect in webview